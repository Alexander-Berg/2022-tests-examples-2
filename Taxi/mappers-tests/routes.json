[
  {
    "expected": [
      {
        "archived_at": "2021-07-06T11:18:03.209129+00:00",
        "is_cyclic": true,
        "name": "moscow.test",
        "path": [
          [
            30.0,
            60.0
          ],
          [
            30.1,
            60.1
          ]
        ],
        "route_id": 123
      }
    ],
    "input": {
      "archived_at": {
        "$date": "2021-07-06T11:18:03.209129Z"
      },
      "is_cyclic": true,
      "name": "moscow.test",
      "path": [
        [
          30.0,
          60.0
        ],
        [
          30.1,
          60.1
        ]
      ],
      "route_id": 123
    }
  }
]
