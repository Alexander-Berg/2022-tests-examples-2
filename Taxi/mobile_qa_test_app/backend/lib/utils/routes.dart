class Routes {
  static const build = '/build/*';
  static const categories = '/categories';
  static const ping = '/ping';
}
