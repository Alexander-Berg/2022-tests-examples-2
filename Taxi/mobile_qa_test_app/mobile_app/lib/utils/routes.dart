import 'package:flutter/material.dart';
import 'package:models/models.dart';

import '../address/ui/page.dart';
import '../app_loader/ui/page.dart';
import '../auth/ui/auth/page.dart';
import '../auth/ui/code/ui/page.dart';
import '../auth/ui/phone/ui/page.dart';
import '../auth/ui/sign_up/ui/page.dart';
import '../build/ui/page.dart';
import '../card/ui/page.dart';
import '../cart/ui/page.dart';
import '../payment/ui/page.dart';
import '../payment_result/ui/page.dart';
import '../profile/ui/page.dart';
import '../shop/ui/category/page.dart';
import '../shop/ui/main/ui/page.dart';
import '../shop/ui/product/page.dart';
import '../support/ui/answer/page.dart';
import '../support/ui/questions/page.dart';

class AppRoutes {
  static const appLoader = '/';
  static const auth = '/auth';
  static const build = '/build';
  static const signUp = '/signUp';
  static const code = '/code';
  static const phone = '/phone';
  static const supportQuestions = '/supportQuestions';
  static const supportAnswer = '/supportAnswer';
  static const shop = '/shop';
  static const card = '/card';
  static const address = '/address';
  static const product = '/product';
  static const cart = '/cart';
  static const paymentResult = '/paymentResult';
  static const payment = '/payment';
  static const profile = '/profile';
  static const category = '/category';
}

Route<dynamic>? onGenerateRoutes(RouteSettings settings) {
  final args = settings.arguments;
  final Widget page;
  switch (settings.name) {
    case AppRoutes.appLoader:
      page = const AppLoaderPage();
      break;
    case AppRoutes.build:
      page = const BuildPage();
      break;
    case AppRoutes.auth:
      page = const AuthPage();
      break;
    case AppRoutes.signUp:
      page = const SignUpPage();
      break;
    case AppRoutes.supportAnswer:
      page = const AnswerPage();
      break;
    case AppRoutes.code:
      page = const CodePage();
      break;
    case AppRoutes.phone:
      page = const PhonePage();
      break;
    case AppRoutes.supportQuestions:
      page = QuestionsPage();
      break;
    case AppRoutes.shop:
      page = const ShopMainPage();
      break;
    case AppRoutes.address:
      page = const AddressPage();
      break;
    case AppRoutes.product:
      page = args == null
          ? const ShopMainPage()
          // ignore: cast_nullable_to_non_nullable
          : ProductPage(args as Product);
      break;
    case AppRoutes.cart:
      page = const CartPage();
      break;
    case AppRoutes.category:
      page = args == null
          ? const ShopMainPage()
          : ShopCategoryPage(args as Category);
      break;
    case AppRoutes.card:
      page = const CardPage();
      break;
    case AppRoutes.paymentResult:
      page = const PaymentResultPage();
      break;
    case AppRoutes.payment:
      page = const PaymentPage();
      break;
    case AppRoutes.profile:
      page = const ProfilePage();
      break;
    default:
      page = Container();
  }

  return MaterialPageRoute<void>(
    builder: (_) => page,
  );
}
