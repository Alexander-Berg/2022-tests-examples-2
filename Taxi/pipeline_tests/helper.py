import collections
import datetime

from js_pipeline.generated import models

from admin_pipeline.generated.service.swagger.models import api
from admin_pipeline.pipeline import common as pipeline_common
from admin_pipeline.pipeline_tests import views as testing_views


class TestingHelper:
    @staticmethod
    async def test_pipeline(request_body, context):
        assert False, 'test_pipeline is not defined in helper'

    def serialize(self, pipeline: models.Pipeline, add_consumer=False) -> dict:
        assert False, 'test_pipeline is not defined in helper'

    @staticmethod
    def __serialize_tests(tests_results):
        # Keys, starting with '$' is forbidden in mongodb
        def __replace_dollar_rec(obj):
            if isinstance(obj, dict):
                return {
                    key.replace('$', '_$'): __replace_dollar_rec(value)
                    for key, value in obj.items()
                }
            if isinstance(obj, list):
                return [__replace_dollar_rec(value) for value in obj]
            return obj

        return __replace_dollar_rec(
            tests_results.body.tests_results.serialize()['tests'],
        )

    async def enumerate_pipeline_tests(self, consumer, pipeline_name, context):
        tests = [
            api.EnumeratedTest(test['_id'], test['name'], test['scope'])
            async for test in testing_views.TestsView.find_many(
                context.mongo.secondary, {'scope': 'global'}, consumer,
            )
        ]
        togglers = await testing_views.ChoosenTestsHelper.try_find_one(
            context.mongo.secondary,
            {'pipeline_name': pipeline_name},
            consumer,
        )
        if togglers:
            tests.extend(
                [
                    api.EnumeratedTest(
                        test['_id'], test['name'], test['scope'],
                    )
                    async for test in testing_views.TestsView.find_many(
                        context.mongo.secondary,
                        {'name': {'$in': togglers['tests']}},
                        consumer,
                    )
                ],
            )
        return api.TestsEnumerateResponse(tests=tests)

    async def enable_pipeline_tests(self, request, context):
        await testing_views.ChoosenTestsHelper.update_one(
            context.mongo,
            {
                '$and': [
                    {'pipeline_name': request.pipeline_name},
                    {'consumer': request.consumer},
                ],
            },
            {'$set': {'tests': request.body.enabled_tests}},
            upsert=True,
        )

    async def get_pipeline_tests_results(self, pipeline_id, consumer, context):
        # intentionally query master
        result = await testing_views.TestsResultsHelper.try_find_one(
            context.mongo,
            {'pipeline_id': pipeline_id},
            consumer,
            for_model=models.PipelineTestsResults,
        )
        if not result:
            return None
        return models.PipelineTestsResults.deserialize(
            testing_views.drop_id(result),
        )

    async def perform_tests(self, pipeline, consumer, context):
        enumerated_tests = await self.enumerate_pipeline_tests(
            consumer, pipeline.name, context,
        )
        tests = [
            await testing_views.TestsView.get(
                enumerated_test.id, consumer, context,
            )
            for enumerated_test in enumerated_tests.tests
        ]
        if not tests:
            return models.PipelineTestsResults(
                created=datetime.datetime.now(), tests=[],
            )

        enumerated_mocks = await testing_views.MocksView.enumerate_mocks(
            consumer, context,
        )
        resources_mocks = collections.defaultdict(dict)
        prefetched_resources_mocks = collections.defaultdict(dict)
        input_mocks = dict()
        for enumerated_mock in enumerated_mocks.resources_mocks:
            mock = await testing_views.MocksView.get(
                enumerated_mock.id, consumer, context,
            )
            resources_mocks[mock.resource][mock.name] = mock.mock
        for enumerated_mock in enumerated_mocks.prefetched_resources_mocks:
            mock = await testing_views.MocksView.get(
                enumerated_mock.id, consumer, context,
            )
            prefetched_resources_mocks[mock.resource][mock.name] = mock.mock
        for enumerated_mock in enumerated_mocks.input_mocks:
            mock = await testing_views.MocksView.get(
                enumerated_mock.id, consumer, context,
            )
            input_mocks[mock.name] = mock.mock

        enumerated_checks = await testing_views.ChecksView.enumerate_checks(
            consumer, context,
        )
        checks = dict()
        for enumerated_check in enumerated_checks.output_checks:
            check = await testing_views.ChecksView.get(
                enumerated_check.id, consumer, context,
            )
            checks[check.name] = check.check.serialize()

        test_request_body = {
            'pipeline': self.serialize(pipeline, add_consumer=True),
            'tests': [test.serialize() for test in tests],
            'resources_mocks': resources_mocks,
            'prefetched_resources_mocks': prefetched_resources_mocks,
            'input_mocks': input_mocks,
            'output_checks': checks,
        }
        tests_results = await self.test_pipeline(test_request_body, context)
        tzinfo = (
            datetime.datetime.now(datetime.timezone.utc).astimezone().tzinfo
        )
        now = datetime.datetime.now().astimezone(tzinfo).isoformat()
        await testing_views.TestsResultsHelper.update_one(
            context.mongo,
            {'$and': [{'pipeline_id': pipeline.id}, {'consumer': consumer}]},
            {
                '$set': {
                    'pipeline_id': pipeline.id,
                    'consumer': consumer,
                    'created': now,
                    'tests': self.__serialize_tests(tests_results),
                },
            },
            upsert=True,
        )
        return await self.get_pipeline_tests_results(
            pipeline.id, consumer, context,
        )

    async def test_pipeline_request(self, request, context):
        pipeline = await pipeline_common.get_by_id(
            request.body.pipeline_id, self, context,
        )
        return await self.perform_tests(pipeline, request.consumer, context)
