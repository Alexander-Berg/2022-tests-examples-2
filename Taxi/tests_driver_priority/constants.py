# generated via tvmknife unittest service -s 200 -d 300 (priority->tags)
TICKET_200_300 = (
    '3:serv:CBAQ__________9_IgYIyAEQrAI:RGXe3QVsvyGEMcOnW4VEqV78XIcLC4XXS7FTy'
    '-iafuLdLR2FY8UmScF7VOCVTh9e3YQfCQ9VsnscVMmdbmiSKDoHZofqGA-rACcc0rdGv9D81'
    'fpualqaL_O6nPEuXtTb7qpxQVjpX1EknAam_FQEKeyEG6KN51rcf4iwgQSsacg'
)

# generated via tvmknife unittest service -s 300 -d 200
TICKET_300_200 = (
    '_3:serv:CBAQ__________9_IgYIrAIQyAE:kqftMeryz0oR__X2DpAGWHga3cFV_y5FmPjY'
    'RASdxB-RY1PgC0aw-YoPQ6ZHU30Ojg07a3EJzWSuJoNdYmpztSiNz4Dn50TXjrDjTsl9CIrU'
    'ugnYYbFFbgrLZR8BmSctwhbObJM37z-IrZgyhM45Lw4kv80MezQ4Xt6r3oazUw'
)

BR_ROOT = 'br_root'
MSK = 'moscow'
SPB = 'spb'
TULA = 'tula'

DEFAULT_ZONES_WHITELIST = {'__default__': True, MSK: True}
DEFAULT_ACCEPT_LANGUAGE = 'en-US, en;q=0.8,ru;q=0.6'
DEFAULT_USER_AGENT = 'Taximeter 9.20 (1234)'
DEFAULT_HEADERS = {
    'Accept-Language': DEFAULT_ACCEPT_LANGUAGE,
    'User-Agent': DEFAULT_USER_AGENT,
}

POLLING_URL = 'v1/polling/priority'
DIAGNOSTICS_URL = 'v1/priority/diagnostics'
SCREEN_URL = 'v1/priority/screen'
VALUE_URL = 'v1/priority/value'
VALUES_URL = 'v1/priority/values'
FLEET_URL = 'fleet/driver-priority/v1/priority/diagnostics'

MSK_COORDS = (37.6, 55.75)
