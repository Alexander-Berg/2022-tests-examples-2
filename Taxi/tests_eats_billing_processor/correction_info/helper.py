def column_format(column_name, column_number, title):
    column = {
        'column_name': column_name,
        'column_number': column_number,
        'title': title,
    }
    return column


def make_doc(doc_id, transaction):
    doc = {'doc_id': doc_id, 'data': {'payments': transaction}}
    return doc


class CorrectionInfoTest:
    def __init__(self):
        self._request = None
        self._response = None
        self._order_nr = None
        self._expect_fail = False
        self._expected_status = 200
        self._docs = None

    def request(self, order_nr):
        self._order_nr = order_nr
        self._request = {'order_nr': self._order_nr}
        return self

    def expect_fail(self, status=500):
        self._expect_fail = True
        self._expected_status = status
        return self

    def response(self, amounts, table_format, corrections, docs=None):
        self._response = {
            'order_nr': self._order_nr,
            'amounts': amounts,
            'table_format': table_format,
            'corrections': corrections,
        }
        self._docs = docs
        return self

    async def run(self, fixtures):
        if self._expect_fail:
            fixtures.billing_reports_mock.should_fail()
        else:
            fixtures.billing_reports_mock.docs(self._docs)

        response = await fixtures.taxi_eats_billing_processor.post(
            '/v1/correction_info', json=self._request,
        )
        assert response.status == self._expected_status
        if not self._expect_fail:
            amounts = self._response['amounts']
            expect_amounts = response.json()['amounts']
            assert len(amounts) == len(expect_amounts)
            for amount in amounts:
                find = False
                for expect_amount in expect_amounts:
                    if amount == expect_amount:
                        find = True
                        break
                assert find is True

            corrections = self._response['corrections']
            expect_corrections = response.json()['corrections']
            assert len(corrections) == len(expect_corrections)
            for correction in corrections:
                find = False
                for expect_correction in expect_corrections:
                    if correction == expect_correction:
                        find = True
                        break
                assert find is True

            table_format = self._response['table_format']
            expect_table_format = response.json()['table_format']
            assert len(table_format) == len(expect_table_format)
            for column in table_format:
                find = False
                for expect_column in expect_table_format:
                    if column == expect_column:
                        find = True
                        break
                assert find is True
