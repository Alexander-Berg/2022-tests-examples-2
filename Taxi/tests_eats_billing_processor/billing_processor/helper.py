import datetime
import json

from testsuite.utils import callinfo


class BillingProcessorTest:
    def __init__(self):
        self._order_nr = '12345'
        self._billing_events = []
        self._columns = ['kind']
        self._expected_transfers = []
        self._should_reschedule = False
        self._expect_fail = False
        self._expected_core_write_back = []

    def for_order_nr(self, order_nr):
        self._order_nr = order_nr
        return self

    def expected_fail(self):
        self._expect_fail = True
        return self

    def insert_billing_event(self, data, input_event_id=1):
        counter = len(self._billing_events) + 1
        self._billing_events.append(
            {
                'input_event_id': input_event_id,
                'order_nr': self._order_nr,
                'external_id': f'auto_{counter}',
                'kind': (
                    'payment'
                    if 'payment' in data.keys()
                    else 'refund'
                    if 'refund' in data.keys()
                    else 'commission'
                ),
                'event_at': datetime.datetime(2021, 3, 26, 17, counter, 0, 0),
                'transaction_date': data['transaction_date'],
                'data': data,
            },
        )
        return self

    def should_reschedule(self):
        self._should_reschedule = True
        return self

    def checking_columns(self, *columns):
        self._columns = columns
        return self

    def expect_transfers(self, *transfers):
        self._expected_transfers.extend(transfers)
        return self

    def expect_core_write_back(self, *events):
        self._expected_core_write_back.extend(events)
        return self

    async def run(self, fixture):
        core_write_back = []

        @fixture.mockserver.json_handler(
            '/eats-billing-storage/billing-storage/create/bulk',
        )
        def _handler(request):
            nonlocal core_write_back
            for item in request.json:
                if item['kind'] == 'CommissionOrder':
                    core_write_back.append(
                        ('commission', item['data']['commission_sum']),
                    )
                else:
                    core_write_back.append(
                        (
                            item['data']['transaction_type'],
                            item['data']['value_amount'],
                            item['data']['commission'],
                        ),
                    )
            response = {'message': 'OK', 'status': 'success'}
            return fixture.mockserver.make_response(json=response, status=200)

        cursor = fixture.pgsql['eats_billing_processor'].cursor()
        for event in self._billing_events:
            data = event['data']
            details = data[event['kind']]
            cursor.execute(
                f"""insert into eats_billing_processor.billing_events
                 (input_event_id, order_nr, external_id, kind, event_at,
                  transaction_date, external_payment_id, client_id,
                  product_type, amount, currency, data)
                 values ({event['input_event_id']}, '{event['order_nr']}',
                         '{event['external_id']}', '{event['kind']}',
                         '{event['event_at']}', '{event['transaction_date']}',
                         '{data['external_payment_id']}',
                         '{data['client']['id']}', '{details['product_type']}',
                          {details['amount']}, '{details['currency']}',
                         '{json.dumps(event['data'])}')""",
            )

        await fixture.stq_runner.eats_billing_processor_billing_processor.call(
            task_id='billing_1',
            kwargs={'order_nr': self._order_nr},
            expect_fail=self._expect_fail,
        )

        try:
            eats_billing_processor_stq = (
                fixture.stq.eats_billing_processor_billing_processor
            )
            call = eats_billing_processor_stq.next_call()
            assert self._should_reschedule
            assert call['id'] == 'billing_1'
        except callinfo.CallQueueEmptyError:
            assert not self._should_reschedule

        cursor.execute(
            f"""select {', '.join(self._columns)}
            from eats_billing_processor.transfers
            where service_order_id='{self._order_nr}'""",
        )
        billing_events = list(cursor)
        assert billing_events == self._expected_transfers

        assert core_write_back == self._expected_core_write_back
