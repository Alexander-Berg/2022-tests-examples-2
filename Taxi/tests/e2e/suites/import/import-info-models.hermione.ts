import {parseSpreadSheetJSON} from 'tests/e2e/helper/parse-spreadsheet';
import {infoModels} from 'tests/e2e/seed-db-map';
import getFixturePath from 'tests/e2e/utils/fixture';
import {describe, expect, it} from 'tests/hermione.globals';

import {ROUTES} from '@/src/constants/routes';

describe('Импорт инфомоделей', function () {
    async function assertUserAttributes(browser: WebdriverIO.Browser) {
        await browser.clickInto('\\#rc-tabs-0-tab-custom-attributes-tab');
        await browser.assertImage('im-user_attributes-table');

        await browser.clickInto(['infomodel-groups-table', 'im_attribute-group_row_attribute_group_code_2']);
        await browser.assertImage('im-user_attributes-table');

        await browser.clickInto(['infomodel-groups-table', 'im_attribute-group_row_attribute_group_code_3']);
        await browser.assertImage('im-user_attributes-table');

        await browser.clickInto(['infomodel-groups-table', 'im_attribute-group_row_undefined']);
        await browser.assertImage('im-user_attributes-table');
    }

    it('Клик на иконку импорта открывает страницу импорта инфомоделей', async function () {
        await this.browser.openPage(ROUTES.CLIENT.INFOMODELS);
        await this.browser.clickInto(['action-bar', 'import-button-link'], {waitRender: true, waitNavigation: true});
        await this.browser.assertImage('base-layout');
    });

    it('Страница импорта инфомоделей открывается по прямой ссылке', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS);
        await this.browser.waitUntilRendered();
        await this.browser.assertImage('base-layout');
    });

    it('Шаблон файла для импорта', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS);
        const link = await this.browser.findByTestId('xlsx-template-link');
        const url = await link.getProperty('href');
        expect(url).to.match(new RegExp('templates/import-info-model-spreadsheet_ru_rev\\d.xlsx'));
    });

    it('Создать инфомодель с необязательными атрибутами', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS);
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('create-info-model-with-not-required-attributes.xlsx')
        );
        await this.browser.waitForTestIdSelectorInDom('import-info-create');
        await this.browser.clickInto(['import-info-create', 'expand-icon'], {waitRender: true});
        await this.browser.assertImage('import-info-create');
        await this.browser.clickInto('submit-button', {waitRender: true});
        await this.browser.waitForTestIdSelectorInDom('import-success-alert');

        await this.browser.openPage(ROUTES.CLIENT.INFOMODELS);
        await this.browser.clickInto('im_row_test_info_model_code', {waitNavigation: true, waitRender: true});
        await this.browser.assertImage('base-layout-content');

        await assertUserAttributes(this.browser);
    });

    it('Изменение всех параметров инфомодели', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS);
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('change-all-info-model-parameters.xlsx')
        );
        await this.browser.waitForTestIdSelectorInDom('import-info-update');
        await this.browser.clickInto(['import-info-update', 'expand-icon'], {waitRender: true});
        await this.browser.assertImage('import-info-update');
        await this.browser.clickInto('submit-button', {waitRender: true});
        await this.browser.waitForTestIdSelectorInDom('import-success-alert');

        await this.browser.openPage(ROUTES.CLIENT.INFOMODEL(infoModels.ru.info_model_code_1_7));
        await this.browser.waitUntilRendered();
        await this.browser.assertImage('base-layout-content');

        await this.browser.clickInto('\\#rc-tabs-0-tab-custom-attributes-tab');
        await this.browser.assertImage('im-user_attributes-table');

        await this.browser.clickInto(['infomodel-groups-table', 'im_attribute-group_row_attribute_group_code_3']);
        await this.browser.assertImage('im-user_attributes-table');
    });

    it('Изменить английское название и описание, Израиль', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS, {region: 'il'});
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('change-info-model-en-title-and-description-il.xlsx')
        );
        await this.browser.waitForTestIdSelectorInDom('import-info-update');
        await this.browser.clickInto('submit-button', {waitRender: true});
        await this.browser.waitForTestIdSelectorInDom('import-success-alert');

        await this.browser.openPage(ROUTES.CLIENT.INFOMODEL(infoModels.il.info_model_code_4_1), {region: 'il'});
        await this.browser.waitUntilRendered();
        await this.browser.assertImage(['translations', 'en']);
    });

    it('Добавить, удалить атрибут и изменить важность, Франция', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS, {region: 'fr'});
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('add-remove-and-change-importance-info-model-fr.xlsx')
        );
        await this.browser.waitForTestIdSelectorInDom('import-info-update');
        await this.browser.clickInto('submit-button', {waitRender: true});
        await this.browser.waitForTestIdSelectorInDom('import-success-alert');

        await this.browser.openPage(ROUTES.CLIENT.INFOMODEL(infoModels.fr.info_model_code_3_8), {region: 'fr'});
        await this.browser.waitUntilRendered();

        await this.browser.clickInto('\\#rc-tabs-0-tab-custom-attributes-tab');
        await this.browser.assertImage('im-user_attributes-table');

        await this.browser.clickInto(['infomodel-groups-table', 'im_attribute-group_row_attribute_group_code_2']);
        await this.browser.assertImage('im-user_attributes-table');
    });

    it('Нельзя удалить атрибут типа boolean из инфомодели', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS);
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('removing-boolean-attribute-from-info-model-is-forbidden.xlsx')
        );
        await this.browser.waitForTestIdSelectorInDom('import-info-not-upload');
        await this.browser.clickInto(['import-info-not-upload', 'expand-icon'], {waitRender: true});
        await this.browser.clickInto('import-cell-tooltip-help-icon', {waitRender: true});
        await this.browser.assertImage('import-info-not-upload');
        await this.browser.waitForTestIdSelectorDisabled(['header-panel', 'submit-button']);
    });

    it('Проверка ссылки при попытке удаления атрибута, присутствующего в товаре, Великобритания', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS, {region: 'gb'});
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('delete-linked-to-product-attribute-from-info-model.xlsx')
        );
        await this.browser.waitForTestIdSelectorInDom('import-info-not-upload');
        await this.browser.clickInto(['import-info-not-upload', 'expand-icon'], {waitRender: true});
        await this.browser.clickInto('import-cell-tooltip-help-icon', {waitRender: true});
        await this.browser.assertImage('import-info-not-upload');
        await this.browser.clickInto(['error-tooltip-body', 'error-tooltip-link'], {waitRender: true});

        const handles = await this.browser.getWindowHandles();
        expect(handles.length).to.equal(2);

        await this.browser.switchToWindow(handles[1]);
        expect(await this.browser.getUrl()).to.match(new RegExp(`${ROUTES.CLIENT.PRODUCTS}\\?filters`));

        await this.browser.waitUntilRendered();
        await this.browser.assertImage('filter-list');

        await this.browser.closeWindow();
        await this.browser.switchToWindow(handles[0]);
    });

    it('Экспорт файла после успешного импорта', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS);
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('purge-info-model-title.xlsx')
        );
        await this.browser.waitForTestIdSelectorInDom('import-info-update');
        await this.browser.clickInto('submit-button', {waitRender: true});
        await this.browser.waitForTestIdSelectorInDom('import-success-alert');

        await this.browser.clickInto('download-xlsx-link', {waitRender: true});

        const file = await this.browser.getDownloadedFile('exported-info-models.xlsx', {purge: true});
        expect(parseSpreadSheetJSON(file)).to.matchSnapshot(this);
    });

    it('Очистить поле title (без локали в заголовке) инфомодели', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS);
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('purge-info-model-title.xlsx')
        );
        await this.browser.waitForTestIdSelectorInDom('import-info-update');
        await this.browser.clickInto(['import-info-update', 'expand-icon'], {waitRender: true});
        await this.browser.assertImage('import-info-update');

        await this.browser.clickInto('submit-button', {waitRender: true});
        await this.browser.waitForTestIdSelectorInDom('import-success-alert');

        await this.browser.openPage(ROUTES.CLIENT.INFOMODEL(infoModels.ru.info_model_code_1_1));
        await this.browser.waitUntilRendered();
        await this.browser.assertImage('translations');
    });

    it('Очистить один из переводов описания, Франция', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS, {region: 'fr'});
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('purge-info-model-description-fr.xlsx')
        );
        await this.browser.waitForTestIdSelectorInDom('import-info-update');
        await this.browser.clickInto(['import-info-update', 'expand-icon'], {waitRender: true});
        await this.browser.assertImage('import-info-update');

        await this.browser.clickInto('submit-button', {waitRender: true});
        await this.browser.waitForTestIdSelectorInDom('import-success-alert');

        await this.browser.openPage(ROUTES.CLIENT.INFOMODEL(infoModels.fr.info_model_code_3_1), {region: 'fr'});
        await this.browser.waitUntilRendered();
        await this.browser.assertImage('translations');
    });

    it('Изменить обязательность атрибутов всеми способами', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS);
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('check-is-important-values.xlsx')
        );
        await this.browser.waitForTestIdSelectorInDom('import-info-update');
        await this.browser.clickInto('submit-button', {waitRender: true});
        await this.browser.waitForTestIdSelectorInDom('import-success-alert');

        await this.browser.openPage(ROUTES.CLIENT.INFOMODEL(infoModels.ru.info_model_code_1_1));
        await this.browser.waitUntilRendered();

        await assertUserAttributes(this.browser);
    });

    it('Уведомление об ошибке в заголовке', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS);
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('import-info-model-with-invalid-header.xlsx')
        );
        await this.browser.waitForTestIdSelectorInDom('alert-warning');
        await this.browser.assertImage('alert-warning');
    });

    it('Нельзя импортировать файл с пустым полем код', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS);
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('import-info-model-with-empty-code.xlsx')
        );
        await this.browser.waitForTestIdSelectorInDom('import-info-not-upload');
        await this.browser.clickInto(['import-info-not-upload', 'expand-icon'], {waitRender: true});
        await this.browser.clickInto('import-cell-tooltip-help-icon', {waitRender: true});
        await this.browser.assertImage('import-info-not-upload');
    });

    it('Игнорирование пустых строк и столбцов в файле', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS);
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('import-info-model-with-empty-columns-and-rows.xlsx')
        );
        await this.browser.waitForTestIdSelectorInDom('import-info-update');
        await this.browser.clickInto(['import-info-update', 'expand-icon'], {waitRender: true});
        await this.browser.assertImage('import-info-update');
    });

    it('Нельзя редактировать скрытую инфомодель с помощью импорта', async function () {
        await this.browser.openPage(ROUTES.CLIENT.IMPORT_INFOMODELS);
        await this.browser.uploadFileInto(
            ['import-upload', 'file-input'],
            getFixturePath('change-hidden-info-model.xlsx')
        );

        await this.browser.waitForTestIdSelectorInDom('import-info-not-upload');
        await this.browser.waitForTestIdSelectorDisabled('submit-button');
        await this.browser.clickInto(['import-info-not-upload', 'expand-icon'], {waitRender: true});
        await this.browser.clickInto('import-cell-tooltip-help-icon', {waitRender: true});
        await this.browser.assertImage('import-info-not-upload');
    });
});
