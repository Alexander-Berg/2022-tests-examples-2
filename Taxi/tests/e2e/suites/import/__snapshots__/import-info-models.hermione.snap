exports[`Импорт инфомоделей Экспорт файла после успешного импорта 1`] = [
  {
    "code": "info_model_code_1_1",
    "title|ru": "",
    "description|ru": "приносят человек возлюбил презирает никакого я некое никого.",
    "attribute": "boolean_attribute_code_0_0",
    "attribute_is_important": "false"
  },
  {
    "code": "",
    "title|ru": "",
    "description|ru": "",
    "attribute": "image_attribute_code_1_0",
    "attribute_is_important": "true"
  },
  {
    "code": "",
    "title|ru": "",
    "description|ru": "",
    "attribute": "multiselect_attribute_code_2_0",
    "attribute_is_important": "false"
  },
  {
    "code": "",
    "title|ru": "",
    "description|ru": "",
    "attribute": "number_attribute_code_3_0",
    "attribute_is_important": "true"
  },
  {
    "code": "",
    "title|ru": "",
    "description|ru": "",
    "attribute": "select_attribute_code_4_0",
    "attribute_is_important": "false"
  },
  {
    "code": "",
    "title|ru": "",
    "description|ru": "",
    "attribute": "string_attribute_code_5_0",
    "attribute_is_important": "true"
  },
  {
    "code": "",
    "title|ru": "",
    "description|ru": "",
    "attribute": "text_attribute_code_6_0",
    "attribute_is_important": "false"
  },
  {
    "code": "",
    "title|ru": "",
    "description|ru": "",
    "attribute": "image",
    "attribute_is_important": "false"
  }
];